drivers/serial/uart_nrf5.c

```diff
@@ -447,7 +447,12 @@ static void uart_nrf5_irq_err_disable(struct device *dev)
 /** Interrupt driven pending status function */
 static int uart_nrf5_irq_is_pending(struct device *dev)
 {
-	return (uart_nrf5_irq_tx_ready(dev) || uart_nrf5_irq_rx_ready(dev));
+	volatile struct _uart *uart = UART_STRUCT(dev);
+
+	return ((uart->INTENSET & UART_IRQ_MASK_TX) &&
```

Comment:
Probably has been worth to cache volatile uart->INTENSET in a local var to optimize this code a bit.

Reply:
Thiis has been rewritten in https://github.com/zephyrproject-rtos/zephyr/pull/7681, so might want to comment there instead